compiler/rustc_hir/src/def.rs

```diff
@@ -796,8 +796,8 @@ impl<Id> Res<Id> {
 pub enum LifetimeRes {
     /// Successfully linked the lifetime to a generic parameter.
     Param {
-        /// Id of the generic parameter that introduced it.
-        param: LocalDefId,
+        /// `LocalDefId` & `Ident` of the generic parameter that introduced it.
+        param: (LocalDefId, Ident),
```

Comment:
Why?

Reply:
To be able to have the lifetime name in the message:
```
warning: elided lifetime has a name
 --> ./test.rs:6:49
  |
6 |     pub fn get_mut<'a>(&'a self, x: &mut u8) -> &mut u8 {
  |                    --                           ^ this elided lifetime gets resolved as `'a`
  |                    |
  |                    lifetime `'a` introduced here
```
Ident contains span and name (as a Symbol). We could get the span back with `tcx.source_span(local_def_id)` or however it's spelled. But not the name: `tcx.def_name(id)` requires `HirId`, not `DefId`, and `tcx.local_def_id_to_hir_id(..)` for some reason makes rustc hang and eat lots of RAM when I try to call it (infinite cycle?).

Reply:
`tcx.item_name` takes a `DefId` and won't need HIR.

Reply:
Oh, sorry, completely overlooked it.

Fixed in d645641862714d02b327e0d9d7d45888a1100e1a